Add ObjectRef:get_children() (#10480)
minetest/minetest: doc/lua_api.txt

       should appear in first person.
* `get_attach()`: returns parent, bone, position, rotation, forced_visible, 
    or nil if it isn't attached.

> * `get_children()`: returns children of the object as a table, or nil if there
>     are none.

Reviewer: ```suggestion
* `get_children()`: returns children of the object as a table, or an empty one if there
    are none.
```
Correspondingly to the suggestion below, here it should be fixed.

Author: The fact it rhymed bugged me, I put "if there aren't any"

> * `get_children()`: returns children of the object as a table, or an empty one
>     if there aren't any.

Reviewer 2: ```suggestion
* `get_children()`: returns a list of `ObjectRefs` that are children of the object
```

Reviewer 3: "* `get_children()`: returns a list of `ObjectRefs` that are attached to the object" might be the most clear.  Also, it could clarify whether it counts children of the children as well.

Author: I think "table" is better suited, as it's also the official term used in Lua. I remember when I first started I wasn't really sure about what a "list" was. I've also checked for `list` and `table` in the lua_api.txt and it's 115 vs 206 occurrences. Also, this looks like nitpicking to me, as three different people (2 core devs and a collaborator) gave me 3 different versions

Reviewer 3: Yeah, table, list, array, etc are used interchangeably in this case.  Table is probably the best.  My main suggestion is to use "attached" instead of "children" since "attached" is the terminology we're currently using, so using "children" in the docs out of the blue may be confusing.  I agree with the parent-child terminology, but I think it's best to stay consistent for now.

(BTW, I'm a native speaker)

Author: Attached though is ambiguous: it doesn't tell the reader about the hierarchy relationship. And I was planning to refactor get/set_attach() anyway to make it consistent with other engines, so...

(I don't know why I thought you were French, my bad!)

Reviewer 3: In that case, just this:

> Also, it should clarify whether it counts children of the children as well.

Because it isn't completely clear.  Children can mean nested children or shallow children in different situations.

Reviewer 4: v-rob/sfan5's versions sound best to me, as a native speaker, I prefer v-rob's as it explicitly mentions attachment. Replacing list with table is fine

Author: I still prefer table but it's 3v1, I'll go with v-rob's then
